src/Css/Style.php

```diff
@@ -1451,6 +1563,27 @@ public function set_prop(string $prop, $val, bool $important = false, bool $clea
             $this->_props_computed[$prop] = $computed;
             $this->_props_used[$prop] = null;
 
+            //TODO: this should be a directed dependency map
+            if (substr($prop, 0, 2) === "--" && !\in_array($prop, $this->_prop_stack, true)) {
+                array_push($this->_prop_stack, $prop);
+                $specified_props = array_filter($this->_props, function($key) {
+                    return \array_key_exists($key, $this->_props_specified);
+                }, ARRAY_FILTER_USE_KEY); // copy existing props filtered by those set explicitly before parsing vars
+                foreach ($specified_props as $specified_prop => $specified_value) {
+                    if (substr($specified_prop, 0, 2) !== "--" || strpos($specified_value, "var($prop") !== false) {
+                        $this->set_prop($specified_prop, $specified_value, \in_array($specified_prop, $this->_important_props, true), true);
```

Comment:
Need to use `isset` here instead of `in_array`, since the property names are stored as the array keys.